### ISSUE TITLE
Validation of invalid XML declarations

### ISSUE THREAD

**Issue by bendrissou** [NaturalIntelligence/fast-xml-parser#616]
### Description
The parser accepts incorrect XML declarations as valid, without raising an error. This is despite enabling the validator option "-v". 

### Input
```xml
<?xml version="1.0" var ?>  <a/>
```
There is an unassigned attribute `var`, which I expected to raise an error. The input is saved in a file named `example.xml`.

### Code
```bash
./src/cli/cli.js -v example.xml
```

### Output
```
{ '?xml': { '@_version': 1 }, a: '' }
```

My question is: Why the XML validator doesn't throw an error?
Please note that xmllint rejects the input example as invalid.

**Comment by bendrissou** [comment]
Hi @amitguptagwl 

Can you please confirm if this is a valid issue?

**Comment by amitguptagwl** [comment]
Sorry I missed your original message. FXP validator checks for syntax error only, any other issue like scheme related, or the one you're highlighted are not covered.
I'm not sure about this issue but you can check it by disallowing boolean attributes. I guess it is by default is set to allow.

**Comment by bendrissou** [comment]
I believe this is a syntax error. The expression does not conform to the XML syntax, or the [XML grammar](https://github.com/antlr/grammars-v4/blob/master/xml/XMLParser.g4). So I assume this is unintended behaviour?

**Comment by amitguptagwl** [comment]
Have you checked by disallowing boolean attributes?

**Comment by bendrissou** [comment]
Hi @amitguptagwl 

As shown above, I have enabled the `-v` option.
I assume the `-v` flag should force the validation of the input before parsing. Is that correct?

**Comment by amitguptagwl** [comment]
Please try it from the code. I'm not sure if the CLI is updated with all the options. This [docs](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/v4/4.XMLValidator.md) can help you.

```js
const {XMLValidator} = require("fast-xml-parser");
const result = XMLValidator.validate(xmlData, {
    allowBooleanAttributes: true
});
```

**Comment by bendrissou** [comment]
Hi @amitguptagwl 

I have tried the code above:

```
#!/usr/bin/env node
const {XMLValidator} = require("../fxp");
const result = XMLValidator.validate('<?xml version="1.0" var ?>  <a/>', {
    allowBooleanAttributes: true
});
console.log(result);
```

Output is:
```
true
```

The XML input was accepted, despite having the unassigned attribute `var`.

**Comment by amitguptagwl** [comment]
As I remember, The validator has no check for xml description tag. same with v5

**Comment by Kristinita** [comment]
**Type: Additional information** ➕

### 1. Summary

fast-xml-parser validator doesn’t check non-valid [**XML declarations**](https://www.tutorialspoint.com/xml/xml_declaration.htm). It would be nice if fast-xml-parser will check declarations as xmllint.

### 2. Non-valid version in the declaration

`KiraXMLNonValidDeclaration.xml`:

```xml
<?xml version="kira-goddess" encoding="UTF-8"?>
	<kira>Goddess!</kira>

```

xmllint output:

```xml
D:\SashaDebugging\KiraFastXMLParser>xmllint --noout --pedantic KiraXMLNonValidDeclaration.xml
KiraXMLNonValidDeclaration.xml:1: parser error : String not closed expecting " or '
<?xml version="kira-goddess" encoding="UTF-8"?>
               ^
KiraXMLNonValidDeclaration.xml:1: parser error : Malformed declaration expecting version
<?xml version="kira-goddess" encoding="UTF-8"?>
               ^
KiraXMLNonValidDeclaration.xml:1: parser error : Blank needed here
<?xml version="kira-goddess" encoding="UTF-8"?>
               ^
KiraXMLNonValidDeclaration.xml:1: parser error : parsing XML declaration: '?>' expected
<?xml version="kira-goddess" encoding="UTF-8"?>
               ^
```

fast-xml-parser output:

```bat
D:\SashaDebugging\KiraFastXMLParser>fxparser -V KiraXMLNonValidDeclaration.xml
true

```

### 3. Non-valid encoding in the declaration

`KiraXMLNonValidDeclaration.xml`:

```xml
<?xml version="1.0" encoding="Kira-Goddess"?>
	<kira>Goddess!</kira>

```

xmllint output:

```xml
D:\SashaDebugging\KiraFastXMLParser>xmllint --noout --pedantic KiraXMLNonValidDeclaration.xml
KiraXMLNonValidDeclaration.xml:1: parser error : Unsupported encoding Kira-Goddess
<?xml version="1.0" encoding="Kira-Goddess"?>

```

fast-xml-parser output:

```bat
D:\SashaDebugging\KiraFastXMLParser>fxparser -V KiraXMLNonValidDeclaration.xml
true

```

Thanks.

**Comment by Kristinita** [comment]
**Status: Not fixed** :crying_cat_face:

The problem still persists for FXP CLI.

### 1. File

`KiraXMLNonValidDeclaration.xml`:

```xml
<?xml version="kira-goddess" encoding="UTF-8"?>
	<kira>Goddess!</kira>

```

### 2. Steps to reproduce

```bat
fxp val --file KiraXMLNonValidDeclaration.xml
```

### 3. Behavior

#### 3.1. Expected

Like xmllint output:

```bat
D:\SashaDebugging\KiraFastXMLParser>xmllint --noout --pedantic KiraXMLNonValidDeclaration.xml
KiraXMLNonValidDeclaration.xml:1: parser error : String not closed expecting " or '
<?xml version="kira-goddess" encoding="UTF-8"?>
               ^
KiraXMLNonValidDeclaration.xml:1: parser error : Malformed declaration expecting version
<?xml version="kira-goddess" encoding="UTF-8"?>
               ^
KiraXMLNonValidDeclaration.xml:1: parser error : Blank needed here
<?xml version="kira-goddess" encoding="UTF-8"?>
               ^
KiraXMLNonValidDeclaration.xml:1: parser error : parsing XML declaration: '?>' expected
<?xml version="kira-goddess" encoding="UTF-8"?>
```

#### 3.2. Current

```bat
D:\SashaDebugging\KiraFastXMLParser>fxp val --file KiraXMLNonValidDeclaration.xml
Valid
```

FXP CLI shows `Valid` for non-valid file.

### 4. Environment

1. Windows 11 [Version 10.0.22621.3085]
1. Node.js 25.0.0
1. FXP CLI 0.0.2

Thanks.
